src/configure/utilities/InputControlProvider.ts

```diff
@@ -0,0 +1,180 @@
+        }
+        return parameters;
+    }
+
+    private _createControls(inputValues: { [key: string]: any}) {
+        for( let input of this._pipelineTemplate.inputs){
+            this._uxInputs.set(input.id, input);
+            var inputUxDescriptor: InputUxDescriptor = null;
+            var inputUxDescriptorValue = this._getInputUxDescriptorValue(input, inputValues);
+            
+            switch(input.inputMode){
+                case InputMode.None: 
+                case InputMode.AzureSubscription:
+                    if(input.type !== InputDataType.Authorization) {
+                        inputUxDescriptor = new InputUxDescriptor(input, inputUxDescriptorValue, ControlType.None);
+                    }
+                    break;
+                case InputMode.TextBox:
+                case InputMode.PasswordBox:
+                    inputUxDescriptor = new InputUxDescriptor(input,inputUxDescriptorValue, ControlType.InputBox);
+                    break;
+                case InputMode.Combo:
+                case InputMode.CheckBox:
+                case InputMode.RadioButtons:
+                    inputUxDescriptor = new InputUxDescriptor(input, inputUxDescriptorValue, ControlType.QuickPick);
+                    break;
+
+            }
+            if(inputUxDescriptor){
+                this._inputUxDescriptors.set(input.id, inputUxDescriptor);
+            }
+        }
+        this._setUxDescriptorsDataSourceInputs();   
+    }
+
+    private _setUxDescriptorsDataSourceInputs(): void {
+        this._uxInputs.forEach((inputDes) => {
+            if (!!inputDes.dataSourceId) {
+                var inputUxDescriptor = this._inputUxDescriptors.get(inputDes.id);
+                inputUxDescriptor.dataSource = DataSourceExpression.parse(inputDes.dataSourceId, this._pipelineTemplate.dataSources);
+
+                if (inputUxDescriptor.dataSource) {
+                    var dependentUxDesciptorArray = this._getInputDependencyArray(inputUxDescriptor, inputUxDescriptor.dataSource.getInputDependencyArray());
+                    if (dependentUxDesciptorArray) {
+                        inputUxDescriptor.dataSourceUxInputs.push(...dependentUxDesciptorArray);
+                    }
+                } else {
+                    throw new Error(`Data source {inputDes.dataSourceId} specified for input {inputDes.id} is not present in pipeline template {this._pipelineTemplate.id}`);
+                }
+            }
+        });
+    }
+
+    private _setupDefaultInputValue(inputUxDescriptor: InputUxDescriptor): void {
+        var inputDes = this._uxInputs.get(inputUxDescriptor.getInputUxDescriptorId());
+        if (!inputDes.defaultValue || !InputUxDescriptorUtility.doesExpressionContainsDependency(inputDes.defaultValue)) {
+            return;
+        }
+        var inputUxDescriptor = this._inputUxDescriptors.get(inputDes.id);
```

Comment:
inputUxDescriptor is already passed as argument

Reply:
removed this line